Library/ViewModels/PledgeViewModel.swift

```diff
@@ -827,7 +827,8 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
 
     trackCheckoutPageViewData
       .observeValues { project, baseReward, rewards, selectedQuantities, refTag, additionalPledgeAmount, pledgeTotal, shippingTotal, context in
-        guard context == .pledge else { return }
+        let allowedPageContexts: [PledgeViewContext] = [.pledge, .update, .updateReward]
+        guard allowedPageContexts.contains(context) else { return }
```

Comment:
pass the context into the analytics method and perform this logic in something like a switch where you can control which method is called.

Reply:
Can you please clarify on this bit, there's only one method that needs to be called.

The `trackCheckoutPaymentPageViewed` method send exactly the same events properties to segment where the  `PledgeViewModel` `context_page` =  `.updatePledge`,  or `.checkout`, i.e If the context of `PledgeViewModel` is`.update` or `.updateReward` the context_page = `.updatePledge`, otherwise it's `.checkout`.

So is there a need to move the logic to KSRAnalytics if we are only working with 2 properties, `.checkout`, and `.updatePledge`?

Reply:
I would move it because this is an unconventional way to perform this kind of logic. In particular, `allowedPageContexts` probably does not need to be here because you can just switch on the context and perform the even conditionally if it falls under the right case.